src/main.ts

```diff
@@ -0,0 +1,137 @@
+import {NotificationMessage} from './NotificationMessage';
+import {EventType} from "./enums";
+import {schemaOnIssue, schemaOnIssueComment, schemaOnPullRequest, schemaOnPush} from "./eventPayloadSchemaBuilder";
+
+const core = require ('@actions/core');
+const path = require('path');
+const https = require('https');
+
+const jsonPath = path.join(__dirname, '..', 'resources', 'notification.json');
+const workflow = process.env.GITHUB_WORKFLOW;
+const repository = process.env.GITHUB_REPOSITORY;
+const branch = process.env.GITHUB_REF;
+const github_event_payload =  process.env.GITHUB_EVENT_PATH || jsonPath;
+const trigger_event_name = process.env.GITHUB_EVENT_NAME;
+
+const msteams_webhook_url = core.getInput('webhook_url');
+const job_status = core.getInput('job_status');
+const event_id = core.getInput('event_id');
+
+export const parsedNotificationMessage = async function parse_event_to_message(event_payload_text: string): Promise<NotificationMessage> {
```

Comment:
Is `Promise<NotificationMessage>` really the correct return type? What if validation fails? I feel like the possibility of failure has to be represented somehow. Most crudely we could have `Promise<NotificationMessage|undefined>`.